### ISSUE TITLE
Can't install on node 0.12

### ISSUE THREAD

**Issue by throrin19** [bnoordhuis/node-iconv#108]
Hi, i try to install this library on Node 0.12 and i have this error : 

```
623 warnings generated.
  CXX(target) Release/obj.target/iconv/src/binding.o
../src/binding.cc:35:7: error: no member named 'Arguments' in namespace 'v8'; did you mean 'v8::internal::Arguments'?
using v8::Arguments;
      ^~~~~~~~~~~~~
      v8::internal::Arguments
/Users/throrin19/.node-gyp/0.12.0/deps/v8/include/v8.h:127:7: note: 'v8::internal::Arguments' declared here
class Arguments;
      ^
../src/binding.cc:67:17: error: calling a protected constructor of class 'v8::HandleScope'
    HandleScope scope;
                ^
/Users/throrin19/.node-gyp/0.12.0/deps/v8/include/v8.h:816:13: note: declared protected here
  V8_INLINE HandleScope() {}
            ^
../src/binding.cc:69:51: error: 'New' is a private member of 'v8::PersistentBase<v8::ObjectTemplate>'
    object_template = Persistent<ObjectTemplate>::New(t);
                                                  ^
/Users/throrin19/.node-gyp/0.12.0/deps/v8/include/v8.h:572:23: note: declared private here
  V8_INLINE static T* New(Isolate* isolate, T* that);
                      ^
../src/binding.cc:69:56: error: too few arguments to function call, expected 2, have 1
    object_template = Persistent<ObjectTemplate>::New(t);
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ^
/Users/throrin19/.node-gyp/0.12.0/deps/v8/include/v8.h:572:3: note: 'New' declared here
  V8_INLINE static T* New(Isolate* isolate, T* that);
  ^
/Users/throrin19/.node-gyp/0.12.0/deps/v8/include/v8config.h:289:20: note: expanded from macro 'V8_INLINE'
# define V8_INLINE inline __attribute__((always_inline))
                   ^
../src/binding.cc:70:20: error: member reference type 'Persistent<v8::ObjectTemplate>' is not a pointer; maybe you meant to use '.'?
    object_template->SetInternalFieldCount(1);
    ~~~~~~~~~~~~~~~^~
                   .
../src/binding.cc:70:22: error: no member named 'SetInternalFieldCount' in 'v8::Persistent<v8::ObjectTemplate, v8::NonCopyablePersistentTraits<v8::ObjectTemplate> >'
    object_template->SetInternalFieldCount(1);
    ~~~~~~~~~~~~~~~  ^
../src/binding.cc:71:22: error: no member named 'New' in 'v8::String'
    obj->Set(String::New("make"),
             ~~~~~~~~^
../src/binding.cc:73:22: error: no member named 'New' in 'v8::String'
    obj->Set(String::New("convert"),
             ~~~~~~~~^
../src/binding.cc:76:5: error: no member named 'New' in 'v8::String'
    EXPORT_ERRNO(EINVAL);
    ^~~~~~~~~~~~~~~~~~~~
../src/binding.cc:75:44: note: expanded from macro 'EXPORT_ERRNO'
#define EXPORT_ERRNO(err) obj->Set(String::New(#err), Integer::New(err))
                                   ~~~~~~~~^
../src/binding.cc:77:5: error: no member named 'New' in 'v8::String'
    EXPORT_ERRNO(EILSEQ);
    ^~~~~~~~~~~~~~~~~~~~
../src/binding.cc:75:44: note: expanded from macro 'EXPORT_ERRNO'
#define EXPORT_ERRNO(err) obj->Set(String::New(#err), Integer::New(err))
                                   ~~~~~~~~^
../src/binding.cc:78:5: error: no member named 'New' in 'v8::String'
    EXPORT_ERRNO(E2BIG);
    ^~~~~~~~~~~~~~~~~~~
../src/binding.cc:75:44: note: expanded from macro 'EXPORT_ERRNO'
#define EXPORT_ERRNO(err) obj->Set(String::New(#err), Integer::New(err))
                                   ~~~~~~~~^
../src/binding.cc:85:9: error: no member named 'Dispose' in 'v8::Persistent<v8::Value, v8::NonCopyablePersistentTraits<v8::Value> >'
    val.Dispose();
    ~~~ ^
../src/binding.cc:86:9: error: no member named 'Clear' in 'v8::Persistent<v8::Value, v8::NonCopyablePersistentTraits<v8::Value> >'
    val.Clear();
    ~~~ ^
../src/binding.cc:91:17: error: calling a protected constructor of class 'v8::HandleScope'
    HandleScope scope;
                ^
/Users/throrin19/.node-gyp/0.12.0/deps/v8/include/v8.h:816:13: note: declared protected here
  V8_INLINE HandleScope() {}
            ^
../src/binding.cc:92:13: error: no member named 'AsciiValue' in 'v8::String'
    String::AsciiValue from_encoding(args[0]);
    ~~~~~~~~^
../src/binding.cc:93:13: error: no member named 'AsciiValue' in 'v8::String'
    String::AsciiValue to_encoding(args[1]);
    ~~~~~~~~^
../src/binding.cc:94:32: error: use of undeclared identifier 'to_encoding'
    iconv_t conv = iconv_open(*to_encoding, *from_encoding);
                               ^
../src/binding.cc:95:55: error: no matching function for call to 'Null'
    if (conv == reinterpret_cast<iconv_t>(-1)) return Null();
                                                      ^~~~
/Users/throrin19/.node-gyp/0.12.0/deps/v8/include/v8.h:6615:19: note: candidate function not viable: requires single argument 'isolate', but no arguments were provided
Handle<Primitive> Null(Isolate* isolate) {
                  ^
../src/binding.cc:97:38: error: member reference type 'Persistent<v8::ObjectTemplate>' is not a pointer; maybe you meant to use '.'?
    Local<Object> t = object_template->NewInstance();
                      ~~~~~~~~~~~~~~~^~
                                     .
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make: *** [Release/obj.target/iconv/src/binding.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1067:12)
gyp ERR! System Darwin 14.1.0
gyp ERR! command "node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/throrin19/Documents/Programmation/Web/goomeoeventsapi/node_modules/iconv
gyp ERR! node -v v0.12.0
gyp ERR! node-gyp -v v1.0.2
gyp ERR! not ok 
npm ERR! Darwin 14.1.0
npm ERR! argv "node" "/usr/local/bin/npm" "install" "iconv"
npm ERR! node v0.12.0
npm ERR! npm  v2.5.1
npm ERR! code ELIFECYCLE

npm ERR! iconv@2.0.7 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the iconv@2.0.7 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the iconv package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls iconv
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/throrin19/Documents/Programmation/Web/goomeoeventsapi/npm-debug.log
```

**Comment by thaumant** [comment]
Same problem here.

**Comment by bnoordhuis** [comment]
Thanks for the bug report.  Fixed in ae0afa6 and published as v2.1.6.

**Comment by throrin19** [comment]
thanks, it works fine
